ts-node-plugin/index.js

```diff
@@ -0,0 +1,38 @@
+const {transform} = require("../dist");
+
+const ModuleKindCommonJS = 1;
+
+/**
+ * ts-node transpiler plugin
+ */
+function create(createOptions) {
+  const {nodeModuleEmitKind} = createOptions;
+  const {module} = createOptions.service.config.options;
+  return {
+    transpile(input, transpileOptions) {
+      const {fileName} = transpileOptions;
+      const transforms = ["typescript"];
+      if (module === ModuleKindCommonJS || nodeModuleEmitKind === "nodecjs") {
+        transforms.push("imports");
+      }
+      if (fileName.endsWith(".tsx") || fileName.endsWith(".jsx")) {
+        transforms.push("jsx");
+      }
+
+      const {code, sourceMap} = transform(input, {
+        transforms,
+        sourceMapOptions: {compiledFilename: fileName},
+        filePath: fileName,
+        preserveDynamicImport: nodeModuleEmitKind === "nodecjs",
+        injectCreateRequireForImportRequire: nodeModuleEmitKind === "nodeesm",
+        disableESTransforms: true,
```

Comment:
The ES transforms in Sucrase are pretty irrelevant these days when targeting node, so I just disable them (rather than try to read `target` from tsconfig), and I'm planning on removing them in an upcoming breaking change.